packages/rules.lua

```diff
@@ -43,25 +43,72 @@ SILE.registerCommand("underline", function (_, content)
   local underlineThickness = font.post.underlineThickness / upem * fontoptions.size
 
   local hbox = SILE.call("hbox", {}, content)
-  local gl = SILE.length() - hbox.width
-  SILE.call("lower", {height = SU.cast("measurement", underlinePosition)}, function()
-    SILE.call("hrule", {width = gl.length, height = underlineThickness})
-  end)
-  SILE.typesetter:pushGlue({width = hbox.width})
+  table.remove(SILE.typesetter.state.nodes) -- steal it back...
```

Comment:
Neither here no there for this PR, but I've thought several times this little "trick" needs to be upgraded to be a first class API of some kind, phantom mode or a reset command with mark points or a push/pop stack or something...

Reply:
Thinking alike, as this is something we do in a lot of places... I'd think of something such as `SILE.helper.box.hbox(content)` that would do the same things as the "hbox" command except the final `table.insert` so one doesn't need the nearly immediate `table.remove` in these cases ;

Might look like a long name, but `SILE.findInTree` could be e.g. relocated to `SILE.helper.ast.findInTree`, where other goodies (e.g. my old #1196 from my initial try at SILE, lol) could also find a place, without polluting the global SILE namespace.

Reply:
Altenately, we could have an `hbox:unpushed` command (build the box in the current context, but doesn't pushes it to the node queue). Less general but maybe clearer. (But that's beyond this PR indeed, just for the discussion)

Reply:
Opened an issue to track this someplace it won't get lost when I close this PR:‌https://github.com/sile-typesetter/sile/issues/1326